server/apps/archived/archived.py

```diff
@@ -78,6 +92,129 @@ def on_create(self, docs):
+                    if not take:
+                        raise bad_req_error(message='One of Take(s) not found in Archived repo')
+
+                    if is_item_in_package(take[0]):
+                        raise bad_req_error(message="Can't kill as one of Take(s) is part of a Package")
+
+        doc['item_id'] = item_id
+        doc[config.ID_FIELD] = id_field
+
+    def delete(self, lookup):
+        """
+        Overriding to handle with Kill workflow in the Archived repo:
+            1. Check if Article has an associated Digital Story and if Digital Story has more Takes.
+               If both Digital Story and more Takes exists then all of them would be killed along with the one requested
+            2. For each article being killed do the following:
+                i.   Apply the Kill Template and create an entry in the published collection.
+                ii.  Query the Publish Queue in Legal Archive and find the subscribers who received the article
+                     previously and create transmission entries in Publish Queue.
+                iii. Change the state of the article to Killed in Legal Archive.
+                iv.  Delete all the published versions from Archived.
+                v.   Send a broadcast email to all subscribers.
+        :param lookup: query to find the article in archived repo
+        :type lookup: dict
+        """
+
+        if app.testing and len(lookup) == 0:
+            super().delete(lookup)
+            return
+
+        # Step 1
+        articles_to_kill = self._find_articles_to_kill(lookup)
+        kill_service = KillPublishService()
+
+        for article in articles_to_kill:
+            # Step 2(i)
+            to_apply_template = {'template_name': 'kill', 'item': article}
+            get_resource_service('content_templates_apply').post([to_apply_template])
+            article = to_apply_template['item']
+            self._remove_and_set_kill_properties(article)
+
+            # Step 2(ii)
+            transmission_details = list(
+                get_resource_service(LEGAL_PUBLISH_QUEUE_NAME).get(req=None,
+                                                                   lookup={'item_id': article[config.ID_FIELD]}))
+
+            if transmission_details:
+                subscriber_ids = [t['_subscriber_id'] for t in transmission_details]
+                query = {'$and': [{config.ID_FIELD: {'$in': subscriber_ids}}]}
+                subscribers = list(get_resource_service('subscribers').get(req=None, lookup=query))
+
+                kill_service.queue_transmission(article, subscribers)
+
+            # Step 2(iii)
+            import_into_legal_archive.apply_async(kwargs={'doc': article})
+
+            # Step 2(iv)
+            super().delete({'item_id': article[config.ID_FIELD]})
+
+            # Step 2(i) - Creating entries in published collection
+            get_resource_service('published').post([article])
```

Comment:
One of the reason behind the change to the expiry process was to avoid some items in production and some items archived. The `archive` collection is the source and `published` collection has versions of the document in `archive`. Hence, in order to efficiently move the item to `archived` collection process looks at the `archive` collection.

As per the current expiry process we have two options:
- write a new task to delete this type of items.
- create an entry in the `archive` collection and let the current expiry process take care of it.